src/main/java/io/gatling/mojo/GatlingMojo.java

```diff
@@ -293,16 +299,16 @@ private void executeGatling(List<String> gatlingJvmArgs, List<String> gatlingArg
     }
   }
 
-  private void recordSimulationResults() throws MojoExecutionException {
+  private void recordSimulationResults(Exception exception) throws MojoExecutionException {
     try {
-      saveListOfNewRunDirectories();
+      saveListOfNewRunDirectories(exception);
       copyJUnitReports();
     } catch (IOException e) {
       throw new MojoExecutionException("Could not record simulation results.", e);
     }
   }
 
-  private void saveListOfNewRunDirectories() throws IOException {
+  private void saveListOfNewRunDirectories(Exception exception) throws IOException {
```

Comment:
I think this is confusing: a method called saveListOfNewRunDirectories that also serializes an Exception.

We have two different concepts here that deserve individual treatment. One is saving the new run directories in `LAST_RUN_FILE`, the other is preserving build errors. For the latter I suggest to add another file _lastBuild.txt_ with its name stored in `LAST_BUILD_FILE`.

Reply:
Regarding files, I disagree. IMHO, having one single file is the best solution:
* using 2 different files would require keeping them in sync. I only see more complexity and can't see the value.
* exceptions are not only compile errors but only runtime ones, eg OOME.

We're keep backward compatibility by not prefixing assertion results.

Regarding method name, I agree we could have something better. Suggestions?

Reply:
Probably we can even split this method in two. Something like:
```java
  private void saveSimulationResultToFile(Exception exception) throws IOException {
    Path resultsFile = resultsFolder.toPath().resolve(LAST_RUN_FILE);

    try (BufferedWriter writer = Files.newBufferedWriter(resultsFile)) {
      saveListOfNewRunDirectories(writer);
      writeExceptionIfExists(writer, exception);
    }
  }

  private void saveListOfNewRunDirectories(BufferedWriter writer) throws IOException {
      for (File directory : directoriesInResultsFolder()) {
        if (isNewDirectory(directory)) {
          writer.write(directory.getName() + System.lineSeparator());
        }
      }
  }
```

Reply:
@slandelle OK.
@pasharik Splitting in two is a good solution.